### PR TITLE
Fix plane collisions

### PR DESCRIPTION
**Pull request by kevinsadi**
Allow collisions with the surfing planes to not be detected as the ground to enable surfing. This works by checking the normal of the object the player is colliding with at the point of contact. If the y value of that collision is close to negative 1, I set the grounded state to true. You can see this logic in GroundedCallback.cpp.

### CODE REVIEW

**Comment by kevinsadi** [comment]
@keptsecret @Bowers-L Movement is a lot closer to correct. 

* Ground collisions are consistent and correct.
* Added test surf map
* modified parameters to have movement feel more true to csgo 
* Surf still does not feel right, I am thinking it has to do with the collision logic, because air strafing definitely feels right (I was able to do some bunnyhopping, which is impossible without airstrafing). I would appreciate more eyes on this.

Here are some references of accurate surf. 

* https://github.com/Maugun/SurfUnity/blob/master/Assets/Scripts/MovementConfig.cs
* https://www.reddit.com/r/Unity3D/comments/dcmvf5/i_replicated_the_source_engines_air_strafing/
* https://github.com/ValveSoftware/source-sdk-2013/blob/56accfdb9c4abd32ae1dc26b2e4cc87898cf4dc1/sp/src/game/shared/gamemovement.cpp

File: src/engine/physics/PhysicsWorld.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace lei3d
 		m_solver = std::make_unique<btSequentialImpulseConstraintSolver>();
 
 		m_dynamicsWorld = std::make_unique<btDiscreteDynamicsWorld>(m_dispatcher.get(), m_overlappingPairCache.get(), m_solver.get(), m_collisionConfiguration.get());
-		m_dynamicsWorld->setGravity(btVector3(0, -10, 0));
+		m_dynamicsWorld->setGravity(btVector3(0, -40, 0));
```
Comment (elsien-dev) [inline src/engine/physics/PhysicsWorld.cpp:24]: Put this in editor on my most recent commit, so I can just add that in.

File: src/engine/core/Camera.cpp [hunk 2]
```diff
@@ -5,11 +5,11 @@
 namespace lei3d
 {
 	Camera::Camera(GLFWwindow* window, float yaw, float pitch, LookMode mode)
-		: m_Yaw{yaw}, m_Pitch{pitch}, m_LookMode{mode}
+		: m_Yaw{ yaw }, m_Pitch{ pitch }, m_LookMode{ mode }
 	{
 		m_FOVDeg = 45.0f;
 		m_NearPlane = 0.1f;
-		m_FarPlane = 700.0f;
+		m_FarPlane = 1400.0f;
```
Comment (elsien-dev) [inline src/engine/core/Camera.cpp:12]: This is so we can see more terrain I'm guessing?

**Review by elsien-dev — APPROVED** [review]

File: src/engine/core/Camera.cpp (hunk 2 above)
Comment (kevinsadi) [reply src/engine/core/Camera.cpp:12]: yep yep, couldn't see the whole scene

File: src/engine/physics/PhysicsWorld.cpp (hunk 1 above)
Comment (kevinsadi) [reply src/engine/physics/PhysicsWorld.cpp:24]: bet

**Pull request merged**